clcache.py

```diff
@@ -1334,7 +1334,8 @@ def processDirect(cache, outputFile, compiler, cmdLine, sourceFile):
             origCmdLine = cmdLine
             stripIncludes = False
             if '/showIncludes' not in cmdLine:
-                cmdLine = ['/showIncludes'] + cmdLine
+                cmdLine = ['/showIncludes']
+                cmdLine.extend(origCmdLine)
                 stripIncludes = True
```

Comment:
*Probably* yes, I didn't think of that. In this case, the suggested solution (either calling `cmdLine.extend` or `cmdLine += `) is actually harmful because it modifies the list argument (which is passed by reference). I.e. the program

```python
def f(xs):
  xs.extend([3,4])

l = [1,2]
f(l)
print len(l)
```

prints '4'. So it was actually intentional to create a new list here, but I tend to agree with QC that it seemed suspicious.